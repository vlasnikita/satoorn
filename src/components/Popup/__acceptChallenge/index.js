import React, { Component, Fragment } from 'react';
import {connect} from "react-redux";

import { acceptChallenge, applyCommission } from 'Fetchers/index'
import { updateChallenge } from "Actions/challenges";
import { showAlert } from "Actions/alert";
import { getOperationHash } from "Utils/index";

import PAYMENT_METHODS from 'Constants/payment_methods'
import CHALLENGE_STATES from 'Constants/challenge_states'
import ALERT_TYPE from 'Constants/alert_type'
import log from '../../../utils/logging';

let hashAccept = null

export class AcceptChallenge extends Component {

    state = {
        selected: Object.keys(PAYMENT_METHODS)[0],
        requisite: '',
        isErrors: [],
        amountWithCommission: 100,
        minimalCommission: null,
        code: 'OK'
    }

    componentDidMount() {
        hashAccept = getOperationHash(`ChallengeAccept${+new Date()}${this.props.acceptingChallenge.id || ''}`)
        this.calculateCommissionSettingState(this.state.selected);
    }

    resolveFormContent = () => {
        if(this.state.code == 'COMMISSION_OVERFLOW') {
            return (
                <div>
                    <p className="PopupChallengeAccept__text">
                            К сожалению, комиссия при данном способе выплате превышает доступную для выплаты сумму :(   
                    </p>
                    <p className="PopupChallengeAccept__text">Поробуйте другой способ выплаты или соберите больше донатов на этот челлендж.</p>    
                </div>    
             )
        } else {
            return this.getForm()   
        }        
    }

    getForm = () => {
        if(this.state.selected === 'CARD') return (
            <Fragment>
                <p className="PopupChallengeAccept__text">Номер карты</p>
                <input
                    id="cardNumber"
                    autoComplete="cc-number"
                    name="cardNumber"
                    className={`PopupChallengeAccept__input ${this.state.isErrors.includes('requisite') && 'PopupChallengeAccept__input_error'}`}
                    type="text"
                    value={this.state.requisite}
                    onChange={this.handleCardNumberChange}
                />
                {this.state.isErrors.includes('requisite') &&
                <p className="PopupChallengeAccept__input_errorMessage">Некорректный номер карты</p>}
            </Fragment>
        )
        else if(this.state.selected === 'QIWI') return (
            <Fragment>
                <p className="PopupChallengeAccept__text">Номер телефона в формате 7XXXXXXXXXX</p>
                <input
                    id="qiwiPhoneNumber"
                    className={`PopupChallengeAccept__input ${this.state.isErrors.includes('requisite') && 'PopupChallengeAccept__input_error'}`}
                    type="text"
                    value={this.state.requisite}
                    onChange={this.handleQiwiWalletChange}
                />
                {this.state.isErrors.includes('requisite') &&
                <p className="PopupChallengeAccept__input_errorMessage">Некорректный номер телефона</p>}
            </Fragment>
        )
    }

    getAmountText = () => {
        return (
            <div className="PopupChallengeAccept__row">
                <p className="PopupChallengeAccept__text">Сумма челленджа с учётом комиссии</p>
                <p className="PopupChallengeAccept__amount">{this.state.amountWithCommission}₽</p>
            </div>
        )
    }

    getMinimalCommissionText = () => {
        return (
            <div className="PopupChallengeAccept__row">
                <p className="PopupChallengeAccept__text">
                    Минимальная комиссия при этом способе выплаты: {this.state.minimalCommission}₽
                </p>
            </div>
        )
    }

    handleCardNumberChange = e => {
        const val = e.target.value.replace(/[^0-9]/g, "").replace(/\W/gi, '').replace(/(.{4})/g, '$1 ').slice(0,22);
        this.setState({ requisite: val, isErrors: this.state.isErrors.filter(el => el !== 'requisite') })
    }

    handleQiwiWalletChange = e => {
        const val = e.target.value.replace(/[^+0-9]/g, "")
        this.setState({ requisite: val, isErrors: this.state.isErrors.filter(el => el !== 'requisite') })
    }

    handleSubmit = () => {
        const { requisite, selected } = this.state
        const filteredRequisite = requisite.split(' ').join('').replace('+', '')
        let isErrors = []

        const creditCardPattern = /^((\d{16})|(\d{18}))$/
        const qiwiWalletPattern = /^(7\d{10})$/

        if(!requisite) isErrors.push('requisite')
        else if(selected === 'CARD' && !(creditCardPattern.test(filteredRequisite))) isErrors.push('requisite')
        else if(selected === 'QIWI' && !(qiwiWalletPattern.test(filteredRequisite))) isErrors.push('requisite')

        if (isErrors.length !== 0) {
            log.error('', isErrors)
            this.setState({isErrors})
        }
        else {
            let amount = this.props.acceptingChallenge.amount
            acceptChallenge(this.props.acceptingChallenge.id, selected, filteredRequisite, amount, hashAccept)
                .then(res => {
                    if(res.code !== ALERT_TYPE.OK){ this.props.handleShowAlert(res.code) }
                    else {
                        hashAccept = getOperationHash(`ChallengeAccept${+new Date()}${this.props.acceptingChallenge.id || ''}`)
                        this.props.handleUpdateChallenge(this.props.acceptingChallenge.id, {
                            state: CHALLENGE_STATES.ACCEPTED,
                            isJustUpdated: true
                        })
                    }
                })
                .catch(error => this.props.handleShowAlert(ALERT_TYPE.INTERNAL_ERROR))

            this.props.handleClose()
        }
    }

    calculateCommissionSettingState(method) {
        applyCommission(this.props.acceptingChallenge.id, method)
            .then(res => this.setState(
                {
                 amountWithCommission: res.amountWithCommission, 
                 minimalCommission: res.minimalCommission,
                 code: res.code
                }
            ))
            .catch(e => log.error('Commission calculation error', e))
    }

    render() {
        return (
            <div className="PopupChallengeAccept">
                {!!this.state.amountWithCommission && this.getAmountText()}

                <div className="PopupChallengeAccept__row">
                    <p className="PopupChallengeAccept__text">Выберите способ выплаты</p>
                    <div className="PopupChallengeAccept__types">
                        {Object.keys(PAYMENT_METHODS).map((key, i) => (
                            <p
                                className={`PopupChallengeAccept__type ${key === this.state.selected && 'PopupChallengeAccept__type_active'}`}
                                key={i}
                                onClick={() => {
                                    this.setState({selected: key, requisite: '' });
                                    this.calculateCommissionSettingState(key)
                                }
                                }
                            >{PAYMENT_METHODS[key]}</p>
                        ))}
                    </div>
                    {!!this.state.minimalCommission && this.getMinimalCommissionText()}
                </div>
                <div className="PopupChallengeAccept__row">{this.resolveFormContent()}</div>

                <button
                    disabled={this.state.code != 'OK'}
                    className="Button Button_purple Popup__submit"
                    onClick={this.handleSubmit}
                >Принять челлендж</button>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({})

const mapDispatchToProps = (dispatch) => {
    return {
        handleShowAlert: type => dispatch(showAlert(type)),
        handleUpdateChallenge: (id, body) => dispatch(updateChallenge(id, body))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(AcceptChallenge)
