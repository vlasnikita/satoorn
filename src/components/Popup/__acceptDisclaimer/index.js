import React, {Component} from 'react';

import {toggleAcceptChallengeOnboardinig} from 'Fetchers/index'

import POPUP_TYPE from 'Constants/popup_type'
import {getSettings} from "Actions/configs";
import {connect} from "react-redux";

class AcceptDisclaimer extends Component {

    state = {
        isChecked: true
    }

    handleSubmit = () => {
        this.props.handleGetSettings()
        this.state.isChecked && toggleAcceptChallengeOnboardinig()
        this.props.handleOpen(POPUP_TYPE.ACCEPT_CHALLENGE, this.props.acceptingChallenge)
    }

    render() {
        return (
            <div className="PopupDisclaimer">
                <p className="PopupDisclaimer__text">
                    –ù–∞–∂–∏–º–∞—è –∫–Ω–æ–ø–∫—É "–°–æ–≥–ª–∞—Å–µ–Ω" –≤—ã –ø–æ–¥—Ç–≤–µ—Ä–∂–¥–∞–µ—Ç–µ —Å–≤–æ—é –≥–æ—Ç–æ–≤–Ω–æ—Å—Ç—å –≤—ã–ø–æ–ª–Ω–∏—Ç—å —á–µ–ª–ª–µ–Ω–¥–∂ —Å–≤–æ–∏—Ö –∑—Ä–∏—Ç–µ–ª–µ–π.<br/><br/>
                    –î–µ–Ω—å–≥–∏ –ø–µ—Ä–µ–≤–µ–¥—É—Ç—Å—è –≤–∞–º —Å—Ä–∞–∑—É –≤ –≤–∏–¥–µ –¥–æ–Ω–∞—Ç–∞. –î–ª—è —ç—Ç–æ–≥–æ –≤ –Ω–æ–≤–æ–º –æ–∫–Ω–µ –Ω—É–∂–Ω–æ –±—É–¥–µ—Ç –≤–≤–µ—Å—Ç–∏ —Å–≤–æ–∏ —Ä–µ–∫–≤–∏–∑–∏—Ç—ã. <br/><br/>
                    –î–æ—Ä–æ–∂–∏—Ç–µ —Å–≤–æ–µ–π –∞—É–¥–∏—Ç–æ—Ä–∏–µ–π ‚Äì –¥–æ–±—Ä–æ—Å–æ–≤–µ—Å—Ç–Ω–æ –≤—ã–ø–æ–ª–Ω—è–π—Ç–µ —á–µ–ª–ª–µ–Ω–¥–∂–∏, –∏ —Ç–æ–≥–¥–∞ –æ–Ω–∞ –±—É–¥–µ—Ç —Ä–∞—Å—Ç–∏. <br/><br/>
                    –£ –Ω–∞—Å –µ—Å—Ç—å –∞—Ä–º–∏—è –º–æ–¥–µ—Ä–∞—Ç–æ—Ä–æ–≤, –∫–æ—Ç–æ—Ä—ã–µ —Ä–∞–∑–±–∏—Ä–∞—é—Ç –∂–∞–ª–æ–±—ã –∑—Ä–∏—Ç–µ–ª–µ–π –Ω–∞ –Ω–µ–≤—ã–ø–æ–ª–Ω–µ–Ω–Ω—ã–µ —á–µ–ª–ª–µ–Ω–¥–∂–∏.<br/>
                    –¢–∞–∫–∏–º –æ–±—Ä–∞–∑–æ–º –º—ã –æ—Å—É—â–µ—Å—Ç–≤–ª—è–µ–º –∫—Ä—É–≥–ª–æ—Å—É—Ç–æ—á–Ω—É—é –ø–æ–¥–¥–µ—Ä–∂–∫—É –Ω–∞—à–∏—Ö –ø–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª–µ–π –∏ –Ω–∞–∫–∞–∑—ã–≤–∞–µ–º –Ω–µ–¥–æ–±—Ä–æ—Å–æ–≤–µ—Å—Ç–Ω—ã—Ö —Å—Ç—Ä–∏–º–µ—Ä–æ–≤ üïµÔ∏è
                </p>
                <div
                    className="CreateOrder__row CreateOrder__row_checkbox PopupDisclaimer__row"
                    onClick={() => this.setState({ isChecked: !this.state.isChecked })}
                >
                    <label className="CreateOrder__input">
                        <input
                            type="checkbox"
                            checked={this.state.isChecked}
                            onChange={() => this.setState({ isChecked: !this.state.isChecked })}
                        />
                        {this.state.isChecked && <span className="CreateOrder__inputChecked"/>}
                    </label>
                    <span className="CreateOrder__inputHint PopupDisclaimer__inputHint">–ë–æ–ª—å—à–µ –Ω–µ –ø–æ–∫–∞–∑—ã–≤–∞—Ç—å</span>
                </div>
                <button
                    className="Button Button_purple Popup__submit"
                    onClick={this.handleSubmit}
                >–Ø—Å–Ω–æ–ø–æ–Ω—è—Ç–Ω–æ</button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        handleGetSettings: () => dispatch(getSettings()),
    }
}

export default connect(null,mapDispatchToProps)(AcceptDisclaimer)