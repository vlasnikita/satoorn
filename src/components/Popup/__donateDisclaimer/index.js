import React, {Component} from 'react';

import {toggleDonateChallengeOnboardinig} from 'Fetchers/index'
import {connect} from "react-redux";
import {getSettings} from "Actions/configs";

class DonateDisclaimer extends Component {

    state = {
        isChecked: true
    }

    handleSubmit = () => {
        this.props.handleGetSettings()
        this.state.isChecked && toggleDonateChallengeOnboardinig()
        this.props.handleClose()
    }

    render() {
        return (
            <div className="PopupDisclaimer">
                <p className="PopupDisclaimer__text">
                    –ù–∞–∂–∏–º–∞—è –∫–Ω–æ–ø–∫—É "–ü–æ–≤—ã—Å–∏—Ç—å" –∏–ª–∏ "–°–æ–∑–¥–∞—Ç—å —á–µ–ª–ª–µ–Ω–¥–∂" –≤—ã –ø–µ—Ä–µ–≤–æ–¥–∏—Ç–µ —É–∫–∞–∑–∞–Ω–Ω—É—é —Å—É–º–º—É –Ω–∞ –ø—Ä–æ–º–µ–∂—É—Ç–æ—á–Ω—ã–π —Å—á—ë—Ç —Å–µ—Ä–≤–∏—Å–∞.<br/><br/>
                    –ï—Å–ª–∏ —Å—Ç—Ä–∏–º–µ—Ä —Å–æ–≥–ª–∞—Å–∏—Ç—Å—è –≤—ã–ø–æ–ª–Ω–∏—Ç—å —á–µ–ª–ª–µ–Ω–¥–∂, –¥–µ–Ω—å–≥–∏ –∞—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ –ø–µ—Ä–µ–≤–µ–¥—É—Ç—Å—è –µ–º—É –∫–∞–∫ –ø—Ä–∏ –æ–±—ã—á–Ω–æ–º –¥–æ–Ω–∞—Ç–µ.<br/><br/>
                    –ï—Å–ª–∏ —Å—Ç—Ä–∏–º–µ—Ä –æ—Ç–∫–∞–∂–µ—Ç—Å—è ‚Äì —Å—É–º–º–∞ –≤–µ—Ä–Ω—ë—Ç—Å—è –≤–∞–º –æ–±—Ä–∞—Ç–Ω–æ –∑–∞ –≤—ã—á–µ—Ç–æ–º –∫–æ–º–∏—Å—Å–∏–∏ –ø–ª–∞—Ç—ë–∂–Ω–æ–≥–æ —Å–µ—Ä–≤–∏—Å–∞.<br/><br/>
                    –ú—ã –ø—Ä–æ—Ç–∏–≤ –∫–æ–º–∏—Å—Å–∏–∏, –Ω–æ –ø–æ–∫–∞ —á—Ç–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ —Å—Ç–æ—Ä–æ–Ω–Ω–µ–≥–æ —Å–µ—Ä–≤–∏—Å–∞ ‚Äì –µ–¥–∏–Ω—Å—Ç–≤–µ–Ω–Ω–∞—è –æ–ø—Ü–∏—è –¥–ª—è –Ω–∞—Å. –ü–æ –º–µ—Ä–µ —Ä–æ—Å—Ç–∞ —Å–µ—Ä–≤–∏—Å–∞ –º—ã —Ä–∞–∑—Ä–∞–±–æ—Ç–∞–µ–º —Å–æ–±—Å—Ç–≤–µ–Ω–Ω—ã–π –ø–ª–∞—Ç—ë–∂–Ω—ã–π –º–æ–¥—É–ª—å, —á—Ç–æ–±—ã –æ—Ç–∫–∞–∑–∞—Ç—å—Å—è –æ—Ç –∫–æ–º–∏—Å—Å–∏–∏ –¥–ª—è –∑—Ä–∏—Ç–µ–ª–µ–π.<br/>
                    –ñ–µ–ª–∞–µ–º –∫–ª–∞—Å—Å–Ω—ã—Ö —Å—Ç—Ä–∏–º–æ–≤ –∏ —Ö–æ—Ä–æ—à–µ–≥–æ –Ω–∞—Å—Ç—Ä–æ–µ–Ω–∏—è! üòâ
                </p>
                <div
                    className="CreateOrder__row CreateOrder__row_checkbox PopupDisclaimer__row"
                    onClick={() => this.setState({ isChecked: !this.state.isChecked })}
                >
                    <label className="CreateOrder__input">
                        <input
                            type="checkbox"
                            checked={this.state.isChecked}
                            onChange={() => this.setState({ isChecked: !this.state.isChecked })}
                        />
                        {this.state.isChecked && <span className="CreateOrder__inputChecked"/>}
                    </label>
                    <span className="CreateOrder__inputHint PopupDisclaimer__inputHint">–ë–æ–ª—å—à–µ –Ω–µ –ø–æ–∫–∞–∑—ã–≤–∞—Ç—å</span>
                </div>
                <button
                    className="Button Button_purple Popup__submit"
                    onClick={this.handleSubmit}
                >–Ø—Å–Ω–æ–ø–æ–Ω—è—Ç–Ω–æ</button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        handleGetSettings: () => dispatch(getSettings()),
    }
}

export default connect(null,mapDispatchToProps)(DonateDisclaimer)