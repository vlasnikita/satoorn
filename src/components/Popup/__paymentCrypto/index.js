import React, {Component, Fragment} from 'react';
import {connect} from "react-redux";

import {toggleDonateChallengeOnboardinig} from 'Fetchers/index'
import {cryptoCurrencies} from 'Constants/currency'
import {getSettings} from "Actions/configs";
import {openPaymentWidget, generateQRCode} from 'Utils';
import {floatMask} from 'Utils/input_masks';
import log from 'Utils/logging';
import {generateFKPaymentLink} from 'Utils/freekassa.js'

class PaymentCrypto extends Component {

    state = {
        isCurrencySet: false,
        isCurrencyListVisible: false,
        activeCurrency: null,
        sum: 0.001,
        tx: null,
        QRData: null
    }

    handleSetActiveCurrency = activeCurrency => {
        if(activeCurrency != this.state.activeCurrency){
            generateQRCode(activeCurrency.address)
            .then(res => this.setState({ activeCurrency, QRData: res }))    
        }
    }

    getActiveCurrency = () => {
        const curr = this.state.activeCurrency

        if(curr && curr.symbol){
            return (
                <div
                    value={curr.symbol}
                    className="Input__selectItem Input__selectItem_active"
                >
                    <img src={curr.icon} />
                    <p>{curr.symbol} ({curr.name})</p>
                    <span className='Input__selectExpand'>‹</span>
                </div>
            )
        } else {
            return <div className="Input__selectItem Input__selectItem_active">--Выберите--</div>
        }
    
    }

    getCurrencyList = () => this.state.isCurrencyListVisible && (
        <div className="Input__selectList">
            {Object.values(cryptoCurrencies).map((item, key)=> (
                <button 
                    key={key}
                    value={item.symbol}
                    className="Input__selectItem"
                    onClick={()=>this.handleSetActiveCurrency(item)}
                >
                    <img src={item.icon} />
                    <p>{item.symbol} ({item.name})</p>
                </button>
            ))}
        </div>
    )
        
    getSteps = (activeCurrency, sum, QRData) => activeCurrency && activeCurrency.symbol && (
        <Fragment>
            <div className="PopupPayment__row PopupPayment__row_step">
                <div className="PopupPayment__column">
                    <div className="PopupPayment__step">
                        <span className='PopupPayment__stepNumber'>1</span>
                        <p className='PopupPayment__stepText'>Переведите <span>{sum}</span> ВТС на адрес <span>{activeCurrency.address}</span></p>
                    </div>
                    <div className="PopupPayment__step">
                        <span className='PopupPayment__stepNumber'>2</span>
                        <p className='PopupPayment__stepText'>А затем укажите хэш транзакции:</p>
                        
                    </div>
                </div>
                <img className='PopupPayment__qr' src={QRData}/>
            </div>
            <div className="Input Input_tx">
                <div className="Input__column">
                    <span className="Input__label">Хэш транзакции</span>
                    <input
                        type="text"
                        className='Input__input Input__input_large'
                        onChange={e => this.setState({ tx: e.target.value })} 
                    />
                </div>
            </div>
        </Fragment>
    )

    render() {
        const {
            isCurrencyListVisible,
            isCurrencySet,
            activeCurrency,
            sum,
            tx,
            QRData
        } = this.state

        return (
            <div className="PopupPayment">
                <div className="Input">
                    <div className="Input__column">
                        <span className="Input__label">Криптовалюта</span>
                        <div 
                            className='Input__input Input__input_medium'
                            onClick={()=> this.setState({isCurrencyListVisible: !isCurrencyListVisible})}        
                        >
                            <div className="Input__select">
                                {this.getActiveCurrency()}
                                {this.getCurrencyList()}
                            </div>
                        </div>
                    </div>
                    {activeCurrency && <div className="Input__column">
                        <span className="Input__label">Сумма в {activeCurrency.symbol}</span>
                        <input
                            className='Input__input Input__input_medium'
                            value={sum}
                            placeholder='0.001'
                            onChange={e => this.setState({ sum: floatMask(e.target.value) })} 
                        />
                    </div>}
                    {/* <div className="Input__column">
                        <span className="Input__label">Сумма в рублях</span>
                        <input 
                            className='Input__input Input__input_short' 
                            placeholder='0.0000001'    
                        />
                    </div> */}
                </div>
                {this.getSteps(activeCurrency, sum, QRData)}
                <button
                    className="Button Button_purple Popup__submit"
                    onClick={this.handleSubmit}
                >Далее</button>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    payment: state.payment
});

const mapDispatchToProps = (dispatch) => {
    return {
        handleGetSettings: () => dispatch(getSettings()),
        handleShowAlert: type => dispatch(showAlert(type)),
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(PaymentCrypto)