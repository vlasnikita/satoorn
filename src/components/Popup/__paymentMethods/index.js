import React, {Component} from 'react';
import {connect} from "react-redux";

import {openPopup} from 'Actions/popup';
import {openPaymentWidget} from 'Utils';
import log from 'Utils/logging';
import {generateFKPaymentLink} from 'Payments/freekassa.js'
import {generateYKPaymentLink} from 'Payments/yookassa.js'
import ALERT_TYPE from "Constants/alert_type";
import POPUP_TYPE from 'Constants/popup_type';

import fiatIcon from 'Static/payments/payments-fiat.gif'
import cryptoIcon from 'Static/payments/payments-crypto.gif'
import newWindowIcon from 'Static/new_window.svg'
import { async } from 'regenerator-runtime';

class PaymentMethods extends Component {

    state = {
        paymentURL: null
    }

    componentDidMount(){
        console.log(this.props.payment)
    }

    // submitFiatPayment = () => {

    //     const { newBidAmount, hash, challenge } = this.props.payment
    //     const { id, title, donationAmount } = challenge

    //     donateChallenge(newBidAmount, id, hash)
    //     .then(res => {
    //         if (res.code !== ALERT_TYPE.OK) {
    //             this.props.handleShowAlert(res.code);
    //         }
    //         else {
    //             const desc = `Повышение челленджа ${title}`;
    //             log.debug(desc, res);
    //             openPaymentWidget(newBidAmount, res.transactionId, desc, true, id, {donationAmount: +donationAmount + newBidAmount});
    //         }
    //     })
    //     .catch(() => this.props.handleShowAlert(ALERT_TYPE.INTERNAL_ERROR));
    // }

    processCryptoPayment = () => {
        const { id, donationAmount } = this.props.payment.challenge
        const paymentURL = generateFKPaymentLink(donationAmount, id)
        this.setState({ paymentURL })
        window.open(paymentURL, '_blank').focus();
    }

    processFiatPayment = async () => {
        const { id, donationAmount, title, streamerLogin } = this.props.payment.challenge
        const paymentURL = await generateYKPaymentLink(id, donationAmount, title, streamerLogin)
        this.setState({ paymentURL })
        window.open(paymentURL, '_blank').focus();
    }

    getSuccessMessage = () => (
        <div className="PopupPayment">
            <h1 className="Popup__title">Платёж открылся в новой вкладке</h1>
            <p class="Popup__subtitle">Если этого не произошло, <a href={this.state.paymentURL} target="_blank">нажмите здесь</a></p>
            <button
                className="Button Button_purple Popup__submit"
                onClick={console.log}
            >Далее</button>
        </div>
    )

    render() {
        if(this.state.paymentURL) return this.getSuccessMessage()
        else return (
            <div className="PopupPayment">
                <h1 className="Popup__title">Выберите способ пополнения</h1>
                <div className="PopupPayment__row">
                    <button 
                        className="PopupPayment__method"
                        onClick={this.processFiatPayment}    
                    >
                        <img src={newWindowIcon} className="PopupPayment__newWindow"/>
                        <img className='PopupPayment__methodIcon' src={fiatIcon}/>
                        <p>Банковская карта</p>
                    </button>
                    <button 
                        className="PopupPayment__method"
                        onClick={this.processCryptoPayment}
                    >
                        <img src={newWindowIcon} className="PopupPayment__newWindow"/>
                        <img className='PopupPayment__methodIcon' src={cryptoIcon}/>
                        <p>Криптовалюта</p>
                    </button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    payment: state.payment,
    profile: state.profile
});

const mapDispatchToProps = (dispatch) => {
    return {
        handleShowAlert: type => dispatch(showAlert(type)),
        handleOpenPopup: (type) => dispatch(openPopup(type))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(PaymentMethods)