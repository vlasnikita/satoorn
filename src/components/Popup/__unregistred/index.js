import React, {Component} from 'react';

import {logAuthButtonClick} from "Utils/metrics";
import {ROUTE_LOGIN} from "Constants/routes";
import twIcon from "Static/social/social-twitch.svg";
import ytIcon from "Static/social/social-youtube.svg";

class Unregistred extends Component {

    render() {
        return (
            <div className="PopupUnregistred">
                <h1 className="Popup__title">Авторизация</h1>
                <div className="PopupUnregistred__list">
                    <a onClick={() => logAuthButtonClick('TWITCH')}
                       id="twitchLoginButton"
                       data-number={2}
                       href={`${ROUTE_LOGIN}/twitch`}
                       className="Level__block Level__block_button">
                        <img className='Level__icon' src={twIcon} alt="" />
                    </a>
                    <a onClick={() => logAuthButtonClick('GOOGLE')}
                       id="googleLoginButton"
                       data-number={3}
                       href={`${ROUTE_LOGIN}/google`}
                       className="Level__block Level__block_button">
                        <img className='Level__icon' src={ytIcon} alt="" />
                    </a>
                </div>
            </div>
        )
    }
}

export default Unregistred