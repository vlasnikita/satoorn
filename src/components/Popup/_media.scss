@media only screen and (max-width: 770px) {
    .Popup{
        &__body{
            width: 100%;
            border-radius: 0;
        }

        &__close{
            left: unset;
            top: -30px;
            right: 5px;
        }
    }
}

@media only screen and (max-width: 600px) {
    .Popup{
        &__body{
            height: 80%;
        }
    }
}