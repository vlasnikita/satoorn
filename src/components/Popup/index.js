import React, { Component } from 'react';
import { connect } from 'react-redux';

import { openPopup, closePopup } from "Actions/popup";
import POPUP_TYPE from 'Constants/popup_type'
import AcceptDisclaimer from "./__acceptDisclaimer";
import DonateDisclaimer from "./__donateDisclaimer";
import AcceptChallenge from "./__acceptChallenge";
import Unregistred from "./__unregistred";
import PaymentMethods from "./__paymentMethods";
import PaymentCrypto from "./__paymentCrypto";

class Popup extends Component {

    getBody = () => {
        const { popup, handleClosePopup, handleOpenPopup } = this.props
        switch (popup.type) {
            case(POPUP_TYPE.ACCEPT_DISCLAIMER):
                return (
                    <AcceptDisclaimer
                        handleOpen={handleOpenPopup}
                        handleClose={handleClosePopup}
                    />
                )
            case(POPUP_TYPE.DONATE_DISCLAIMER):
                return (
                    <DonateDisclaimer handleClose={handleClosePopup}/>
                )
            case(POPUP_TYPE.ACCEPT_CHALLENGE):
                return (
                    <AcceptChallenge
                        handleClose={handleClosePopup}
                        acceptingChallenge={popup.acceptingChallenge}
                    />
                )
            case(POPUP_TYPE.UNREGISTRED):
                return (
                    <Unregistred handleClose={handleClosePopup} />
                )
            case(POPUP_TYPE.PAYMENT_METHODS):
                return (
                    <PaymentMethods handleClose={handleClosePopup} />
                )
            case(POPUP_TYPE.PAYMENT_CRYPTO):
                return (
                    <PaymentCrypto handleClose={handleClosePopup} />
                )
        }
    }

    render() {
        return (
            <div className="Popup">
                <div className="Popup__body">
                    <button onClick={this.props.handleClosePopup} className="Popup__close">&#x2715;</button>
                    {this.getBody()}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    popup: state.popup
});

const mapDispatchToProps = (dispatch) => {
    return {
        handleClosePopup: () => dispatch(closePopup()),
        handleOpenPopup: (type, acceptingChallenge) => dispatch(openPopup(type, acceptingChallenge))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(Popup);