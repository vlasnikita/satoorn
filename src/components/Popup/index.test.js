import React from 'react'
import 'jest-specific-snapshot'
import renderer from 'react-test-renderer'
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter } from 'react-router-dom';

import {unmountComponentAtNode} from "react-dom";
import Popup from "./index";
import POPUP_TYPE from "Constants/popup_type";

const mockStore = configureMockStore();

const initialState = {
    popup: {
        opened: false,
        type: undefined
    }
}

it('renders correctly', () => {
    const store = mockStore(initialState);
    const tree = renderer.create(
        <BrowserRouter>
            <Provider store={store}>
                <Popup/>
            </Provider>
        </BrowserRouter>
    )

    expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly Popup.HELP', () => {
    const initialState2 = {
        popup: {
            opened: false,
            type: POPUP_TYPE.HELP
        }
    }
    const store2 = mockStore(initialState2);
    const tree = renderer
        .create(
            <Provider store={store2}>
                <Popup />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSpecificSnapshot('./__specifics__/help_popup_renders.specific');
});

it('renders correctly Popup.ACCEPT_DISCLAIMER', () => {
    const initialState3 = {
        popup: {
            opened: false,
            type: POPUP_TYPE.ACCEPT_DISCLAIMER
        }
    }
    const store3 = mockStore(initialState3);
    const tree = renderer
        .create(
            <Provider store={store3}>
                <Popup />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSpecificSnapshot('./__specifics__/accept_popup_renders.specific');
});