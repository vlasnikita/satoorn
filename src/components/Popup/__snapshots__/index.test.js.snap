// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Popup"
>
  <div
    className="Popup__body"
  >
    <button
      className="Popup__close"
      onClick={[Function]}
    >
      âœ•
    </button>
  </div>
</div>
`;
